src/LiFi.unit.spec.ts

```diff
@@ -210,4 +259,28 @@ describe('LIFI SDK', () => {
       )
     })
   })
+
+  describe('Should pick up gas from signer estimation', () => {
+    it('should pick up gas estimation from signer', async () => {
+      const route = buildRouteObject({
+        step,
+      })
+
+      await lifi.executeRoute(signer, route)
+
+      expect(signer.sendTransaction).toHaveBeenCalledWith({
+        gasLimit: '125000',
+        // TODO: Check the cause for gasLimit being outside transactionRequest. Currently working as expected in widget
+        transactionRequest: {
+          chainId: 137,
+          data: '0xdata',
+          from: '0x552008c0f6870c2f77e5cC1d2eb9bdff03e30Ea0',
+          gasLimit: '682701',
+          gasPrice: '0x27c01c1727',
+          to: '0x1231DEB6f5749EF6cE6943a275A1D3E7486F4EaE',
+          value: '0x0600830dbc7f5bf7',
+        },
+      })
+    })
+  })
```

Comment:
@Abhikumar98 I just realized that we test functionality for `RouteExecutionManager` here and LIFI class just extends it. Does it make sense to move these tests to RouteExecutionManager.unit.spec.ts?

Reply:
Yes we can do that. That'd would be better